protocols/bftcosi/bftcosi.go

```diff
@@ -97,23 +102,46 @@ type ProtocolBFTCoSi struct {
 
 	// finale signature that this BFTCoSi round has produced
 	finalSignature *BFTSignature
+
+	// our index in the Roster list
+	index int
+	// prepareSignature is the signature generated during the prepare phase
+	// This signature is adapted according to the exceptions that occured during
+	// the prepare phase.
+	prepareSignature []byte
+	// commitSignature is the signature generated during the commit phase
+	// This signature is adapted according to the exceptions that occured during
+	// the commit phase.
+	commitSignature []byte
```

Comment:
Hmm, no really good suggestion, but one idea would be to move the above fields into a (private) struct and all the temp* fields, too. I'm sure there would be nicer ways to make the code more readable but I would need to think more about that.